main_test.go

```diff
@@ -48,3 +51,91 @@ func TestCafeWhenOk(t *testing.T) {
 		assert.Equal(t, http.StatusOK, response.Code)
 	}
 }
+
+func TestCafeCount(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(mainHandle))
+	defer server.Close()
+
+	city := "moscow"
+	requests := []struct {
+		count int
+		want  int
+	}{
+		{0, 0},
+		{1, 1},
+		{2, 2},
+		{100, -1},
+	}
+
+	for _, r := range requests {
+		url := fmt.Sprintf("%s/cafe?city=%s&count=%d", server.URL, city, r.count)
+		resp, err := http.Get(url)
+
+		require.NoError(t, err)
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		_ = resp.Body.Close()
+
+		body := string(bodyBytes)
+		var cafes []string
+
+		if body != "" && body != "," {
```

Comment:
Может тут имелось в виду так
```suggestion
		if body != "" && !strings.Contains(body, ",") {
```